### DISCUSSION TITLE
Orphaned /blog/authors page

### DISCUSSION THREAD

**Discussion by jcubic** [facebook/docusaurus#10662] [Q&A]
I just got notification from Ahrefs that I have orphaned page (no incoming links) to page /blog/authors. It's inside the sitemap, but there are no links to it in my website. It seems that it was added but not linked from the template.

I don't report a bug, because it requires to create reproduction and it's a lot of effort. I also don't see how I'm supposed to create reproduction for this.

Is there a way to remove it from the sitemap and the website? My blog authors have external links there are no profile pages like you have on Docusaurus website.

**Comment by slorber** [comment]
But this authors page is not empty right?

Not sure this can be considered as a bug. Somehow this behavior is by design: you may want to have this page to exist despite not explicitly linking to it. 

By the way it's similar to docs and blog: if you remove navbar/footer, you can have docs and a blog that are not accessible from your Docusaurus layout, and that's fine.


Now, if you really want to get rid of that page, we can provide an option such as `blogAuthorsListComponent: null | false`

---

We can eventually look for a better heuristic to turn this feature on/off automatically. 

If none of the global authors has a dedicated Docusaurus author page, maybe we could turn it off. But maybe some users will appreciate having an overview of all the blog authors, and be able to check their external links, descriptions, or whatever? 

I'm afraid that kind of heuristic is not ideal because in many cases having just an author index page can be useful. See for example: https://react.dev/community/team (not using Docusaurus, but similar)

If we remove that behavior, we should provide a way to turn it on again.

**Reply by jcubic** [reply]
I'm fine with the page, but it should be included somewhere. I'm not sure how it looks like right with the latest version, but when I setup the Docusaurus for the first time, it was not linked anywhere. Which means it's dead code that it's only linked from the sitemap.

**Reply by slorber** [reply]
Yes but how is this different from this blog archive page that also exists for a very long time?

https://docusaurus.io/blog/archive

The truth is: whatever you don't link to is an orphan. It's not dead code, it's accessible and crawlable by other means.

The solution to this problem could simply be adding links to it in our init template in the footer, but technically you can already do this yourself, and it clutters your footer with extra items. Or maybe we should link to it from another place, for example somewhere in the blog sidebar.

Note that most sites will have a way to "discover" that page, because it appears on the author page (if you have any): 

![CleanShot 2024-11-18 at 17 58 10](https://github.com/user-attachments/assets/7bff9d54-6841-402d-b8d1-1d1a0a6aa34e)

**Reply by Josh-Cena** [reply]
BTW the orphan pages of the blog plugin are why my orphan pages checker doesn't work out of the box, so I'm definitely +1 to making them linked from somewhere on the blog landing page. (RSS feeds too)

**Reply by slorber** [reply]
What's this orphan pages checker and how does it work? What does it mean to work out of the box? You mean reporting 0 error on a new site?

What do you mean by "(RSS feeds too)"? 🤔

**Reply by Josh-Cena** [reply]
This thing: https://github.com/facebook/docusaurus/pull/7601 Yes it should report 0 errors on a new site (or with any site that only has control over the MD content).

I'm wondering whether we could add a link to the RSS feed on the landing page, just so people know that this blog provides RSS.

**Reply by slorber** [reply]
Ah I see

My idea was to also have blog sidebar items like we have for general navbar, including items for blog author pages, blog archive, RSS feeds and tags. Not sure it would be a beautiful design but at least it's flexible and solves the problem.

I wonder where else we could show all these items otherwise. Above the paginated posts is a bit awkward to design IMHO.

**Reply by jcubic** [reply]
Where I can find this global /blog/ page? I don't see any file in blog directory besides blog posts and on my project I don't have list of blog authors, only blog posts. Where is this configured? I just tried docusaurus.new, and I see authors, from where they come from?

I have external links as URLs, and I don't have individual profile pages.

**Reply by Josh-Cena** [reply]
`/blog/`, `/blog/authors`, `/blog/archive`, and several other index pages are not written by you. They are synthesized by Docusaurus using our collected metadata. You can customize their appearance via swizzling, but whether you can disable each one depends on whether we expose an option to.

**Reply by jcubic** [reply]
So what I need to do in order to have authors on my blog page? I have authors data, but I have only a single author.

**Reply by slorber** [reply]
The `/blog/authors` page is created by default as soon as you have one `authors.yml` author, which our init template has:

https://tutorial.docusaurus.io/blog/authors/

Note we do the same for tags. For example our init template has blog tags:

https://tutorial.docusaurus.io/blog/tags/

But it doesn't have any docs tags (although it could) and this page doesn't exist by default: 

https://tutorial.docusaurus.io/docs/tags/

**Reply by jcubic** [reply]
I talk about the /blog page that have a list of authors. When the list of authors are created?

**Reply by slorber** [reply]
I don't understand, where do you see a list of authors on this page?

https://docusaurus.io/blog

The suggestion I made was to actually display a link to that list of authors [`/blog/authors`](https://docusaurus.io/blog/authors) somewhere in the blog sidebar. This would be a new thing, not something that already exists.

**Reply by Josh-Cena** [reply]
> I talk about the /blog page that have a list of authors. When the list of authors are created?

It's created programmatically by Docusaurus. If you are asking if there's a way for you to control whether it's generated, the answer is "no at the moment".

**Reply by Josh-Cena** [reply]
In all fairness we could easily use the same `archiveBasePath &&` config logic for authors pages too.

**Reply by slorber** [reply]
yes, we already have `authorsBasePath` so it would just be adding support for a null value
